### ISSUE TITLE
Topic show action touches topic (updates updated_at)

### ISSUE THREAD

**Issue by benjaminwood** [rubysherpas/forem#696]
[This line](https://github.com/rubysherpas/forem/blob/master/app/models/forem/concerns/viewable.rb#L21) touches the topic (changes it's `updated_at`) when the topics/show action is hit. If you are like me and you're implementing fragment caching in your views and you are using the `updated_at` column to determine your cache_key, the cache will be invalidated on every request. I don't see any reason that it should touch the record.

I know the gem isn't really maintained at the moment, but I can create a PR to change this behavior if somebody with merge power can confirm the current behavior is not desired.

Thanks!! :smile:

**Comment by ghost** [comment]
> I know the gem isn't really maintained at the moment

A lot of important sites use this gem. I think the main reason it isn't maintained as frequently as before is because there's just not much more left to do.

**Comment by benjaminwood** [comment]
@hakkiplaten indeed, we have a good size project that has been using the gem for years. I only mentioned that because of the message in the readme:

> I am currently looking for a new maintainer for this gem. I am no longer doing any more work on this myself. Please contact me@ryanbigg.com if you'd like to take over this project.

**Comment by benjaminwood** [comment]
Btw, @hakkiplaten, what are your thoughts on topics getting touched on every show action?

**Comment by c2169000** [comment]
> we have a good size project that has been using the gem for years.

@benjaminwood same here. I don't want to switch to Thredded.

Could you explain a bit to us younger folks why and how you're doing fragment caching in Forem, and how it has improved your app's performance?

**Comment by c2169000** [comment]
> we have a good size project that has been using the gem for years.

@benjaminwood same here. I don't want to switch to Thredded.

Assuming you've found a solution, could you explain a bit to us younger folks why and how you're doing fragment caching in Forem, and how it has improved your app's performance?

**Comment by benjaminwood** [comment]
Hey @c2169000 basically just implementing what DHH describes here: https://signalvnoise.com/posts/3113-how-key-based-cache-expiration-works

Caching has made all the difference for us. Forem was becoming quite slow before we did the caching work. Literally 10+ seconds to load some topics. Now pretty much everything loads in milliseconds to 1.5 seconds for larger topics/forums.

**Comment by glebm** [comment]
@c2169000 Just want to point out that Thredded provides a migration that copies all the data over from Forem so switching might easier than you thought:

https://github.com/thredded/thredded/wiki/Migrate-from-Forem
